### PR TITLE
Core: recontextualize `CollectionState.collect`

### PR DESCRIPTION
**Pull request by alwaysintreble**
## What is this fixing or adding?
renames the `event` kwarg to `prevent_sweep` and removes the "event" behavior that allows for forced collection of non progression.

## How was this tested?
Only by running tests. needs more testing, but if there are any bugs that occur from this, I'd consider them bugs that have been there and were just hidden.

### CODE REVIEW

**Comment by alwaysintreble** [comment]
It wouldn't surprise me at all if most/all of the stardew uses are either completely incorrect or unneeded, given `event` translates to ID-less but still progression items for most people, which is not the intended usage here.

File: worlds/alttp/test/dungeons/TestDungeon.py [hunk 1]
```diff
@@ -54,7 +54,7 @@ def run_tests(self, access_pool):
 
                 for item in items:
                     item.classification = ItemClassification.progression
-                    state.collect(item, event=True)  # event=True prevents running sweep_for_events() and picking up
+                    state.collect(item, prevent_sweep=True)  # event=True prevents running sweep_for_events() and picking up
```
Comment (Jouramie) [inline worlds/alttp/test/dungeons/TestDungeon.py:57]: This comment seems outdated now

**Review by Jouramie — APPROVED** [review]
Stardew change LGTM.  Have you looked if any world use non progression event?

**Comment by NewSoupVi** [comment]
I agree with this fully. I'm not sure either if anything currently relies on the removed code block, but also agree that nothing should ever work like that.

**Review by NewSoupVi — APPROVED** [review]

**Pull request merged**